packages/mui-base/src/Tooltip/TooltipArrow.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { FloatingArrow } from '@floating-ui/react';
+import type { ArrowOwnerState, ArrowProps } from './Tooltip.types';
+import { resolveClassName } from '../utils/resolveClassName';
+import { useTooltipContentContext } from './TooltipContentContext';
+import { evaluateRenderProp } from '../utils/evaluateRenderProp';
+import { useRenderPropForkRef } from '../utils/useRenderPropForkRef';
+
+function defaultRender(props: any, ownerState: ArrowOwnerState) {
+  return (
+    <FloatingArrow
+      context={ownerState.floatingContext}
+      tipRadius={ownerState.tipRadius}
+      staticOffset={ownerState.staticOffset}
+      d={ownerState.d}
+      strokeWidth={1}
+      stroke="transparent"
+      {...props}
+    />
+  );
+}
+
+/**
+ * The tooltip arrow caret element.
+ *
+ * Demos:
+ *
+ * - [Tooltip](https://mui.com/base-ui/react-tooltip/)
+ *
+ * API:
+ *
+ * - [TooltipArrow API](https://mui.com/base-ui/react-tooltip/components-api/#tooltip-arrow)
+ */
+const TooltipArrow = React.forwardRef(function TooltipArrow(
```

Comment:
I'm not sure how I feel about the API of this component. It diverges from our other components as it describes visual aspects of the arrow (dimensions, etc.). It also leaks implementation details.
However, undoubtely it is convenient for common use cases. I'll play with it a bit more and we can discuss the shape of it.